### PR TITLE
Add migration for removing duplicate wallet data. Migrate all address columns to case insensitive format.

### PR DESCRIPTION
**Pull request by IvanIvanoff**
This migration:

1. Converts ProjectEthAddress' address to case insensitive.
2. Removes all duplicates from LatestEthWalletData that were introduced with a previous PR. In this previous PR we started downcasing the address before importing it, so it was doubled. Remove all records for which the address is seen more than once (compared case insensitive) and leave the one which is lowercased (for consistency).

3. Do the same with the corresponding BTC tables.

### CODE REVIEW

File: priv/repo/migrations/20180215105804_remove_duplicates.exs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  import Ecto.Query
+  alias Sanbase.Repo
+
+  alias Sanbase.Model.{
+    LatestEthWalletData,
+    LatestBtcWalletData
+  }
+
+  def up do
+    execute("CREATE EXTENSION IF NOT EXISTS citext WITH SCHEMA public;")
```
Comment (valo) [inline priv/repo/migrations/20180215105804_remove_duplicates.exs]: This extension seems to be supported by RDS, but still we should be aware that extensions could be problematic in RDS.

File: priv/repo/migrations/20180215105804_remove_duplicates.exs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    Repo.insert_all(LatestBtcWalletData, filtered_btc_data)
+  end
+
+  defp count_address(map, addr) do
+    Enum.count(map, fn %LatestEthWalletData{address: address} ->
+      String.downcase(addr) == String.downcase(address)
+    end)
+  end
+
+  defp count_address(map, addr) do
```
Comment (valo) [inline priv/repo/migrations/20180215105804_remove_duplicates.exs]: This function will be shadowed by the previous one

**Comment by valo** [comment]
Can we test this on a dump of staging first?

File: priv/repo/migrations/20180215105804_remove_duplicates.exs (hunk 2 above)
Comment (IvanIvanoff) [reply priv/repo/migrations/20180215105804_remove_duplicates.exs]: Yeah, the pattern match should be in function declaration. We should update our `seeds.exs` to be able to catch search errors earlier.

File: priv/repo/migrations/20180215105804_remove_duplicates.exs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    migrate_btc()
+  end
+
+  def down do
+    # We delete duplicated data - that cannot be rollbacked
+  end
+
+  defp migrate_eth() do
+    alter table(:project_eth_address) do
```
Comment (MilanRusev) [inline priv/repo/migrations/20180215105804_remove_duplicates.exs]: Isn't it possible that in project_eth_address we also have duplicates with lower and upper cases? If yes this statement will probably fail if we don't handele them first. The same for `migrate_eth()`

File: priv/repo/migrations/20180215105804_remove_duplicates.exs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      modify(:address, :citext, null: false)
+    end
+
+    latest_eth_data = Repo.all(LatestEthWalletData)
+
+    filtered_eth_data =
```
Comment (MilanRusev) [inline priv/repo/migrations/20180215105804_remove_duplicates.exs]: What is the benefit of fetching all data in elixir and reinserting it again to just using a single delete sql statement?

File: priv/repo/migrations/20180215105804_remove_duplicates.exs (hunk 3 above)
Comment (IvanIvanoff) [reply priv/repo/migrations/20180215105804_remove_duplicates.exs]: I manually checked both prod and stage for such cases

File: priv/repo/migrations/20180215105804_remove_duplicates.exs (hunk 4 above)
Comment (IvanIvanoff) [reply priv/repo/migrations/20180215105804_remove_duplicates.exs]: None I am aware of, it's just easier to implement for me

**Pull request merged**